README.md

```diff
@@ -6,4 +6,39 @@
 [![Build Status](https://travis-ci.org/fluture-js/concurrify.svg?branch=master)](https://travis-ci.org/fluture-js/concurrify)
 [![Code Coverage](https://codecov.io/gh/fluture-js/concurrify/branch/master/graph/badge.svg)](https://codecov.io/gh/fluture-js/concurrify)
 
-Does nothing yet.
+Turn non-concurrent [FantasyLand 3][FL3] Applicatives concurrent.
+
+## Usage
+
+```js
+//The concurrify takes four arguments, explained below.
+const concurrify = require('concurrify');
+
+//We load the Type Representative of the Applicative we want to transform.
+const Future = require('fluture');
+
+//We create a "zero" instance and a "race" function for "Alternative".
+const zero = Future(() => {});
+const alt = Future.race;
+
+//We create an override "ap" function that works in parallel.
+const ap = (mx, mf) => Future.both(mx, mf).map(([x, f]) => f(x));
+
+//Calling concurrify with these arguments gives is a new Type Representative.
+const ConcurrentFuture = concurrify(Future, zero, alt, ap);
+
+//We can use our type as such:
+ConcurrentFuture(Future.of(1)).sequential.fork(console.error, console.log);
+```
+
+## API
+
+```hs
+concurrify :: (Applicative m, Alternative n)
+           => (TypeRep m, m a, (m a, m a) -> m a, (m a, m (a -> b)) -> m b)
+           -> n m
```

Comment:
yeh I think you need to use `Use ExistentialQuantification or GADTs` extension for that to work, (but I've almost never written haskell so not sure), but as I know sometimes instead of doing that, people move that constraint to instance declarations, for example:

```haskell
data Concurrent f b = Concurrent {sequential :: f b}

instance Applicative a => Functor (Concurrent a) where
    map f (Concurrent g)  = Concurrent (f <$> g)

-- instead of

data Concurrent f b =  Applicative a => Concurrent {sequential :: f b}

instance Functor (Concurrent a) where
    map f (Concurrent g)  = Concurrent (f <$> g)
```